### PR TITLE
Add local_activity_execution_failed metric

### PR DESCRIPTION
**Pull request by Quinn-With-Two-Ns**
add local_activity_execution_failed to align with java

for https://github.com/temporalio/sdk-features/issues/167

### CODE REVIEW

File: internal/common/metrics/constants.go [hunk 1]
```diff
@@ -52,7 +52,8 @@ const (
 
 	LocalActivityTotalCounter           = TemporalMetricsPrefix + "local_activity_total"
 	LocalActivityCanceledCounter        = TemporalMetricsPrefix + "local_activity_canceled"
```
Comment (cretz) [inline internal/common/metrics/constants.go:54]: Should we also have `local_activity_execution_cancelled`?

Comment (Quinn-With-Two-Ns) [reply internal/common/metrics/constants.go:54]: yeah that's missing to I'll add in this PR

**Review by cretz — APPROVED** [review]
Saw that this came from @tsurdilo's SDK features request. Would like @bergundy's confirmation that this is an acceptable approach before merging. Adding extra metrics is ok with me.

**Comment by Quinn-With-Two-Ns** [comment]
In go we are still missing https://docs.temporal.io/references/sdk-metrics#activity_execution_cancelled in the java sdk, but this PR should at least align the go sdk with what our docs say

File: internal/common/metrics/constants.go [hunk 2]
```diff
@@ -50,12 +50,14 @@ const (
...
 
-	LocalActivityTotalCounter           = TemporalMetricsPrefix + "local_activity_total"
-	LocalActivityCanceledCounter        = TemporalMetricsPrefix + "local_activity_canceled"
-	LocalActivityFailedCounter          = TemporalMetricsPrefix + "local_activity_failed"
-	LocalActivityErrorCounter           = TemporalMetricsPrefix + "local_activity_error"
-	LocalActivityExecutionLatency       = TemporalMetricsPrefix + "local_activity_execution_latency"
-	LocalActivitySucceedEndToEndLatency = TemporalMetricsPrefix + "local_activity_succeed_endtoend_latency"
+	LocalActivityTotalCounter             = TemporalMetricsPrefix + "local_activity_total"
+	LocalActivityCanceledCounter          = TemporalMetricsPrefix + "local_activity_canceled" // Deprecated: Use LocalActivityExecutionCanceledCounter instead.
+	LocalActivityExecutionCanceledCounter = TemporalMetricsPrefix + "local_activity_execution_cancelled"
```
Comment (bergundy) [inline internal/common/metrics/constants.go:55]: Are we using 2 `l`s in Java?

File: internal/common/metrics/constants.go [hunk 3]
```diff
@@ -50,12 +50,14 @@ const (
...
+	LocalActivityFailedCounter            = TemporalMetricsPrefix + "local_activity_failed" // Deprecated: Use LocalActivityExecutionFailedCounter instead.
+	LocalActivityExecutionFailedCounter   = TemporalMetricsPrefix + "local_activity_execution_failed"
+	LocalActivityErrorCounter             = TemporalMetricsPrefix + "local_activity_error"
```
Comment (bergundy) [inline internal/common/metrics/constants.go:58]: What's the difference between these 2?
I don't see `local_activity_error` in our docs.

File: internal/common/metrics/constants.go (hunk 2 above)
Comment (bergundy) [reply internal/common/metrics/constants.go:55]: Yes: https://github.com/temporalio/sdk-java/blob/c5f0ebbe8a5cd52ee018809ecdd466379b5a7339/temporal-sdk/src/main/java/io/temporal/worker/MetricsType.java

**Comment by bergundy** [comment]
Overall LGTM, I'm fine duplicating some metrics to keep backwards compatibility.
Let's add an issue to track removal of duplicates after a while (duration not defined at the moment).
Also, please don't mark the issue as closed, we need to fix the Core metrics too.
I'd want to have sdk-features validation before we close.

File: internal/common/metrics/constants.go (hunk 2 above)
Comment (Quinn-With-Two-Ns) [reply internal/common/metrics/constants.go:55]: yes, go and java match

File: internal/common/metrics/constants.go (hunk 3 above)
Comment (Quinn-With-Two-Ns) [reply internal/common/metrics/constants.go:58]: I didn't touch `local_activity_error ` in this PR. Looks like it counts how many times local activities panic

**Comment by Quinn-With-Two-Ns** [comment]
Issue to deprecate in the future: https://github.com/temporalio/sdk-go/issues/968

File: internal/common/metrics/constants.go (hunk 3 above)
Comment (bergundy) [reply internal/common/metrics/constants.go:58]: And what does `local_activity_execution_failed` track?

Comment (Quinn-With-Two-Ns) [reply internal/common/metrics/constants.go:58]: `local_activity_execution_failed ` tracks if the local activity fails. If the activity panics both are incremented

**Pull request merged**